### DISCUSSION TITLE
Erfaringer med Jøkul og TestComplete?

### DISCUSSION THREAD

**Discussion by wkillerud** [fremtind/jokul#2764] [Annet]
Lurer på om dere vet om noen _pain points_ som testerene i teamene deres har med Jøkul og TestComplete? Jeg ser vi for eksempel har typen [DataTestAutoId](https://github.com/fremtind/jokul/blob/49405ecb0a77aa1486aaea4274b04c0b49961e4f/packages/core/src/index.ts#L12) som er i bruk i _noen_ komponenter som et [custom attribute for TestComplete](https://support.smartbear.com/testcomplete/docs/app-testing/web/general/object-identification/using-custom-attributes.html), men ikke alle komponentene bruker den. Og av komponentene som bruker det er det ikke alle elementer inni som kan få en sånn ID. Og er det attributtet _egentlig etablert_ på tvers av organisasjonen, eller er det bare noe ett eller to team bruker?

**Comment by ghost** [comment]
Eg har lagt til data-testautoid på Jøkul-komponentar der SeOpp trengte det. Eg hadde nok ikkje heilt peiling på kva eg dreiv på med, og ser no at fleire av komponentane eg har lagt det til på er det overflødig fordi restProps blir delegert vidare. I tillegg har SeOpp omstrukturert bruken av data-testautoid, og me bruker det no ikkje på samtlige Jøkul-komponentar eg la det til på.

Eg vil tru at i dei aller fleste tilfeller blir det overflødig å ha data-testautoid som ein definert prop på komponentar i Jøkul så lenge man kan legge på rest-props.

Eg veit at Jo Emil har brukt data-testautoid på Jøkul-komponentar. Kanskje du har noko å legge til @joms?

**Reply by joms** [reply]
Sist jeg styret noe særlig med dette var det noe manglende støtte i Jøkul, og at flere komponenter ikke la på rest-props. Litt usikker på hvordan det er nå, men det burde kanskje være et gjennomgående mønster at komponenter støttet rest-props sånn at man enkelt kan legge på ting som dette?

**Comment by kennidenni** [comment]
Vi legger til f.eks data-testid der vi trenger det slik at vi ikke mikser de med data-testautoid.
Vi bruker `data-testid={`${name}-question`}`  for radio-buttons, `data-testid={`${name}-datepicker`}` for date-picker og `data-testid={`${name}-input`}` for andre inputs (både select og textinput)
